From a14d8e1026ad7208ea5851585078a75cb3f5f3be Mon Sep 17 00:00:00 2001
From: Jiri Marsicek <jiri.marsicek@gmail.com>
Date: Sun, 11 Nov 2018 21:34:37 +0100
Subject: [PATCH] Fix x64 arch for Fedora 29

---
 Make.rules | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Make.rules b/Make.rules
index 903a5a4..b8be192 100644
--- a/Make.rules
+++ b/Make.rules
@@ -2,6 +2,7 @@ EFISIGNED = $(patsubst %.efi,%-signed.efi,$(EFIFILES))
 MANPAGES = $(patsubst doc/%.1.in,doc/%.1,$(wildcard doc/*.1.in))
 HELP2MAN = help2man
 ARCH	 = $(shell uname -m | sed 's/i.86/ia32/;s/arm.*/arm/')
+EFIARCH	 = $(shell uname -m | sed 's/i.86/ia32/;s/arm.*/arm/;s/x86_64/x64/')
 ifeq ($(ARCH),ia32)
 ARCH3264 = -m32
 else ifeq ($(ARCH),x86_64)
@@ -13,17 +14,17 @@ ARCH3264 =
 else
 $(error unknown architecture $(ARCH))
 endif
-INCDIR	   = -I$(TOPDIR)include/ -I/usr/include/efi -I/usr/include/efi/$(ARCH) -I/usr/include/efi/protocol
+INCDIR	   = -I$(TOPDIR)include/ -I/usr/include/efi -I/usr/include/efi/$(ARCH) -I/usr/include/efi/protocol -I/usr/include/efi/$(EFIARCH)
 CPPFLAGS   = -DCONFIG_$(ARCH)
 CFLAGS	   = -O2 -g $(ARCH3264) -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check
 LDFLAGS	   = -nostdlib
-CRTOBJ		= crt0-efi-$(ARCH).o
+CRTOBJ		= crt0-efi-$(EFIARCH).o
 CRTPATHS	= /lib /lib64 /lib/efi /lib64/efi /usr/lib /usr/lib64 /usr/lib/efi /usr/lib64/efi /usr/lib/gnuefi /usr/lib64/gnuefi
 CRTPATH		= $(shell for f in $(CRTPATHS); do if [ -e $$f/$(CRTOBJ) ]; then echo $$f; break; fi; done)
 CRTOBJS		= $(CRTPATH)/$(CRTOBJ)
 # there's a bug in the gnu tools ... the .reloc section has to be
 # aligned otherwise the file alignment gets screwed up
-LDSCRIPT	= elf_$(ARCH)_efi.lds
+LDSCRIPT	= elf_$(EFIARCH)_efi.lds
 LDFLAGS		+= -shared -Bsymbolic $(CRTOBJS) -L $(CRTPATH) -L /usr/lib -L /usr/lib64 -T $(LDSCRIPT)
 LOADLIBES	= -lefi -lgnuefi $(shell $(CC) $(ARCH3264) -print-libgcc-file-name)
 FORMAT		= --target=efi-app-$(ARCH)
-- 
2.17.2

