From 41153aaea0c8be5107ba2740db169b1a8b9b9f26 Mon Sep 17 00:00:00 2001
From: Jiri Marsicek <jiri.marsicek@gmail.com>
Date: Sun, 11 Nov 2018 21:03:10 +0100
Subject: [PATCH] Fix EFI_ARCH and gnu-efi crt paths for Fedora

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index bce6f0e..5527cd4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -64,13 +64,13 @@ PKG_CHECK_MODULES(uuid, uuid,
     AC_MSG_ERROR([libuuid (from the uuid package) is required]))
 
 dnl gnu-efi headers require extra include dirs
-EFI_ARCH=$(uname -m | sed 's/i.86/ia32/;s/arm.*/arm/')
+EFI_ARCH=$(uname -m | sed 's/i.86/ia32/;s/arm.*/arm/;s/x86_64/x64/')
 AM_CONDITIONAL(TEST_BINARY_FORMAT, [ test "$EFI_ARCH" = "arm" -o "$EFI_ARCH" = "aarch64" ])
 
 ##
 # no consistent view of where gnu-efi should dump the efi stuff, so find it
 ##
-for path in /lib /lib64 /usr/lib /usr/lib64 /usr/lib32 /lib/efi /lib64/efi /usr/lib/efi /usr/lib64/efi; do
+for path in /lib /lib64 /usr/lib /usr/lib64 /usr/lib32 /lib/efi /lib64/efi /usr/lib/efi /usr/lib64/efi /usr/lib/gnuefi /usr/lib64/gnuefi; do
     if test -e $path/crt0-efi-$EFI_ARCH.o; then
        CRTPATH=$path
     fi
-- 
2.17.2

